### PR TITLE
Increase snap coverage

### PR DESCRIPTION
**Pull request by MirijaHa**
This PR makes manual change checks obsolete by adding a snapshot test for loans and equity each, using extreme values for the model parameters.
Also skip_slow_tests is added as option.

@maurolepore, there is 3 things I am currently not happy with:

- all snaps end up in 1 file as it is currently implemented. This could be solved by using the  argument variant for expect_snapshot, but this does not solve that:
- the printed snaps to the console clutter the console so that one cannot see details of warnings/fails on test in the console. I am not sure if one can solve it, because the snap appears to be somehow generated from the output, wrapping the expect_snapshot in invisible does not work
- each time the test is run now warnings are generated (presumably with the info that a new snap was added). I ruled out it is about the test content by simply copy pasting the already existing snapshot test

Before I spend more time on this, do you have recommendations?

For now this is FYI for @jacobvjk , i will explain in detail how to setup and use this later :).

### CODE REVIEW

**Review by maurolepore — APPROVED** [review]
LGTM. See my nit comments. 

In response to your bullets:

1. In general I'm okay with all snapshots in a single file. When something fails the output prints just the relevant bits. In this case, and because you (and I) are bothered by it, I take it as a signal that  the design of `run_stress_test()` can improve. If we had `stress_test_bonds()` and `stress_test_loans()`, each would get its own file Then we would have _snaps/stress_test_bonds.md and  _snaps/stress_test_loans.md -- i.e. in separate files.

2. Does the output print every single time you run `devtools::test()`? For me it works as expected, where output is produced only in these cases:

    a. when the snapshot is created
    b. when the new snapshot is different from the one stored.
    c. when I run the test interactively -- not via `devtools::test()`. (In interactive mode testthat does not know which file to work with, so all it can do is print the output for you to inspect it).

3. Each time you get a warning? That's unexpected. Once a snapshot is created there should be no warning if the snapshot did not change, or an error if it changed.

Because we're discussing things you experience interactively, it may help so see what you see -- maybe via a video or call.

File: tests/testthat/test-run_stress_test.R [hunk 1]
```diff
@@ -23,3 +25,67 @@ test_that("with bonds output is unchanged and names have the suffix '_arg'", {
...
+
+  in_agnostic <- Sys.getenv("ST_AGNOSTIC")
+  in_specific <- Sys.getenv("ST_SPECIFIC_EQUITY")
+  out <- tempfile()
+  fs::dir_create(out)
+
+  suppressed_console_output <- suppressWarnings(capture.output(
+    results <- run_stress_test("equity",
+      input_path_project_agnostic = in_agnostic,
+      input_path_project_specific = in_specific,
```
[the remark this answers is not in the document]
Comment (MirijaHa) [reply tests/testthat/test-run_stress_test.R:76]: done

File: tests/testthat/test-run_stress_test.R [hunk 2]
```diff
@@ -23,3 +25,67 @@ test_that("with bonds output is unchanged and names have the suffix '_arg'", {
...
+  in_specific <- Sys.getenv("ST_SPECIFIC_LOANS")
+  out <- tempfile()
+  fs::dir_create(out)
+
+  suppressed_console_output <- suppressWarnings(capture.output(
+    results <- run_stress_test("loans",
+      input_path_project_agnostic = in_agnostic,
+      input_path_project_specific = in_specific,
+      output_path = out,
+      lgd_senior_claims = as.numeric(dplyr::filter(stress_test_arguments, name == "lgd_senior_claims")$min),
```
[the remark this answers is not in the document]
Comment (MirijaHa) [reply tests/testthat/test-run_stress_test.R:46]: done

**Comment by MirijaHa** [comment]
> LGTM. See my nit comments.
> 
> In response to your bullets:
> 
> 1. In general I'm okay with all snapshots in a single file. When something fails the output prints just the relevant bits. In this case, and because you (and I) are bothered by it, I take it as a signal that  the design of `run_stress_test()` can improve. If we had `stress_test_bonds()` and `stress_test_loans()`, each would get its own file Then we would have _snaps/stress_test_bonds.md and  _snaps/stress_test_loans.md -- i.e. in separate files.
> 2. Does the output print every single time you run `devtools::test()`? For me it works as expected, where output is produced only in these cases:
>    a. when the snapshot is created
>    b. when the new snapshot is different from the one stored.
>    c. when I run the test interactively -- not via `devtools::test()`. (In interactive mode testthat does not know which file to work with, so all it can do is print the output for you to inspect it).
> 3. Each time you get a warning? That's unexpected. Once a snapshot is created there should be no warning if the snapshot did not change, or an error if it changed.
> 
> Because we're discussing things you experience interactively, it may help so see what you see -- maybe via a video or call.

I still pushed changes for 2 nit comments.

On the rest unfortunately I still observed said behavior so I set up a call to show u what is going on :)

File: tests/testthat/helper.R [hunk 3]
```diff
@@ -28,3 +28,40 @@ test_create_target_directory <- function(path = NULL,
...
+#' @examples
+#' get_st_argument("term")
+#' # Same
+#' get_st_argument("term", "default")
+#'
+#' get_st_argument("term", "min")
+#'
+#' typeof(get_st_argument("company_exclusion"))
+#' typeof(get_st_argument("asset_type"))
+#' @noRd
```
Comment (maurolepore) [inline tests/testthat/helper.R:55]: Here is an alternative implementation of the helper:

``` r
get_st_argument("term")
#> [1] 2
# Same
get_st_argument("term", "default")
#> [1] 2

get_st_argument("term", "min")
#> [1] 1

typeof(get_st_argument("company_exclusion"))
#> [1] "logical"
typeof(get_st_argument("asset_type"))
#> [1] "character"
```

<sup>Created on 2021-12-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Comment (MirijaHa) [reply tests/testthat/helper.R:44]: used yours now

**Pull request merged**